Downsample/Downsample.cpp

```diff
@@ -138,17 +138,26 @@ Eigen::MatrixXf gridSum(const Eigen::MatrixXf &r, int downsamplingFactor)
 
 Eigen::MatrixXf lineSum(const Eigen::MatrixXf &matrix, int downsamplingFactor)
 {
+    Eigen::Map<Eigen::MatrixXf> reshapedMatrixMap(const_cast<float *>(matrix.data()),
+                                                  downsamplingFactor,
+                                                  (int)matrix.rows() * (int)matrix.cols() / downsamplingFactor);
     Eigen::MatrixXf reshapedMatrix =
-        matrix.reshaped(downsamplingFactor, (int)matrix.rows() * (int)matrix.cols() / downsamplingFactor);
+        Eigen::Map<Eigen::MatrixXf>(reshapedMatrixMap.data(), reshapedMatrixMap.rows(), reshapedMatrixMap.cols());
 
     std::function<float(float)> nan_to_zero_functor = nanToZero;
 
     reshapedMatrix = reshapedMatrix.unaryExpr(nan_to_zero_functor);
```

Comment:
Do we actually modify some of the data in `reshapedMatrix` which we casted away `const` for here?

Reply:
Yes we replace NaNs with Zeroes to be able to sum data.

Reply:
So the underlying data on which we replace NaNs with Zeros is actually the data which we previously cast away `const` from? If that's the case, we have to change this. We can't take a `const` parameter to a function, cast away `const` from it, and then mutate the data.